test/e2e/operator_test.go

```diff
@@ -91,7 +92,31 @@ func TestVersionReporting(t *testing.T) {
 		t.Errorf("failed to get deployment %s/%s: %v", deployment.Namespace, deployment.Name, err)
 	}
 
-	patch := []byte(`{"spec": {"template": {"spec": {"containers": [{"name":"dns-operator","env":[{"name":"RELEASE_VERSION","value":"0.0.1-test"}]}]}}}}`)
+	numContainers := len(deployment.Spec.Template.Spec.Containers)
+	if numContainers != 1 {
+		t.Errorf("expected operator deployment to contain 1 container but found %d", numContainers)
+	}
+
+	var curVersion string
+	for _, env := range deployment.Spec.Template.Spec.Containers[0].Env {
+		if env.Name == "RELEASE_VERSION" {
+			curVersion = env.Value
+			break
+		}
+	}
+	if len(curVersion) == 0 {
+		t.Errorf("env RELEASE_VERSION not found in the operator deployment")
+	}
+
+	defer func() {
+		patch := []byte(fmt.Sprintf(`{"spec": {"template": {"spec": {"containers": [{"name":"dns-operator","env":[{"name":"RELEASE_VERSION","value":"%s"}]}]}}}}`, curVersion))
+		if err := sdk.Patch(deployment, types.StrategicMergePatchType, patch); err != nil {
+			t.Fatalf("failed to restore dns operator to old release version: %v", err)
+		}
+	}()
```

Comment:
Could the `sdk.Patch` below mutate the object and then return a non-nil error value? If not, we could put this `defer` statement with the restorative `sdk.Patch` after the initial `sdk.Patch`.  But what you have is safer if `sdk.Patch` could possibly fail even after patching the object.